app/views/questionnaire.py

```diff
@@ -468,8 +455,8 @@ def get_page_title_for_location(schema, current_location, metadata, answer_store
         group = schema.get_group(schema.get_group_by_block_id(block['id'])['id'])
         page_title = '{group_title} - {survey_title}'.format(group_title=group['title'], survey_title=schema.json['title'])
     elif block['type'] == 'Question':
-        first_question = next(schema.get_questions_for_block(block))
-        question_title = get_question_title(first_question, answer_store, schema, metadata)
+        question = choose_question_to_display(block, schema, metadata, answer_store)
+        question_title = question.get('title')
```

Comment:
Similar to the comment about choosing a variant after rendering the block, we should really be determining which question will be displayed in one place. Having said that, this might be easier once the new version of piping is implemented.